fundamentals/fundamentals-collections/src/main/scala/izumi/functional/IzEitherAggregations.scala

```diff
@@ -0,0 +1,101 @@
+import scala.language.implicitConversions
+
+trait IzEitherAggregations {
+  @inline implicit final def EitherBiAggregate[L, R, Col[x] <: IterableOnce[x]](
+    col: Col[Either[NonEmptyList[L], R]]
+  ): EitherBiAggregate[L, R, Col] = new EitherBiAggregate(col)
+  @inline implicit final def EitherBiFlatAggregate[L, R, Col[x] <: IterableOnce[x], Col2[x] <: IterableOnce[x]](
+    col: Col[Either[NonEmptyList[L], Col2[R]]]
+  ): EitherBiFlatAggregate[L, R, Col, Col2] = new EitherBiFlatAggregate(col)
+  @inline implicit final def EitherBiFind[Col[x] <: IterableOnce[x], T](col: Col[T]): EitherBiFind[Col, T] = new EitherBiFind(col)
+  @inline implicit final def EitherBiFoldLeft[Col[x] <: IterableOnce[x], T](col: Col[T]): EitherBiFoldLeft[Col, T] = new EitherBiFoldLeft(col)
+  @inline implicit final def EitherBiMapAggregate[Col[x] <: IterableOnce[x], T](col: Col[T]): EitherBiMapAggregate[Col, T] = new EitherBiMapAggregate(col)
+  @inline implicit final def EitherBiFlatMapAggregate[Col[x] <: IterableOnce[x], T](col: Col[T]): EitherBiFlatMapAggregate[Col, T] = new EitherBiFlatMapAggregate(col)
+}
+
+object IzEitherAggregations extends IzEitherAggregations {
+
+  final class EitherBiAggregate[L, R, Col[x] <: IterableOnce[x]](private val col: Col[Either[NonEmptyList[L], R]]) extends AnyVal {
+    /** `sequence` with error accumulation */
+    def biAggregate(implicit b: Factory[R, Col[R]]): Either[NonEmptyList[L], Col[R]] = {
+      new IzEither.EitherBiAggregate(col)
+        .biAggregate(b, _.toList)
+        .left.map(NonEmptyList.unsafeFrom)
+    }
+
+    /** `sequence_` with error accumulation */
+    def biAggregateVoid: Either[NonEmptyList[L], Unit] = {
+      new IzEither.EitherBiAggregate(col)
+        .biAggregateVoid(_.toList)
+        .left.map(NonEmptyList.unsafeFrom)
+    }
+  }
+
+  final class EitherBiFlatAggregate[L, R, Col[x] <: IterableOnce[x], Col2[x] <: IterableOnce[x]](private val result: Col[Either[NonEmptyList[L], Col2[R]]]) extends AnyVal {
+    /** `flatSequence` with error accumulation */
+    def biFlatAggregate(implicit b: Factory[R, Col[R]]): Either[NonEmptyList[L], Col[R]] = {
+      new IzEither.EitherBiFlatAggregate(result)
+        .biFlatAggregate(b, _.toList)
+        .left.map(NonEmptyList.unsafeFrom)
+    }
+  }
+
+  final class EitherBiFind[Col[x] <: IterableOnce[x], T](private val s: Col[T]) extends AnyVal {
+    def biFind[E](predicate: T => Either[NonEmptyList[E], Boolean]): Either[NonEmptyList[E], Option[T]] = {
+      IzEither.EitherBiFind(s)
+        .biFind(predicate(_).left.map(_.toList))
+        .left.map(NonEmptyList.unsafeFrom)
+    }
+  }
+
+  final class EitherBiFoldLeft[Col[x] <: IterableOnce[x], T](private val col: Col[T]) extends AnyVal {
+    /** monadic `foldLeft` with error accumulation */
+    def biFoldLeft[L, A](z: A)(op: (A, T) => Either[NonEmptyList[L], A]): Either[NonEmptyList[L], A] = {
+      IzEither.EitherBiFoldLeft(col)
+        .biFoldLeft(z)((a,t) => op(a,t).left.map(_.toList))
```

Comment:
Same as above, `biFoldLeft` is not actually list specific and can be generalized:

```scala
    def biFoldLeft[E, A](z: A)(op: (A, T) => Either[E, A]): Either[E, A] = {
```

You may extract conversions for biFind/biFoldLeft into a separate trait, e.g. `IzEitherTraversals` and inherit it in IzEitherAggregations to re-export the methods without writing the conversion methods twice.

```suggestion
trait IzEither extends IzEitherTraversals
trait IzEitherAggregations extends IzEitherTraversals

trait IzEitherTraversals {
  implicit final def EitherBiTraversals[Col[x] <: IterableOnce[x], T](col: T): EitherBiTraversals[Col, T] = new EitherBiTraversals(col)
}

final class EitherBiTraversals[Col[x] <: IterableOnce[x], T](private val col: T) extends AnyVal {
  def biFind...
  def biFoldLeft...
}
```